collector/analyzer/tcpmetricanalyzer/tcp_analyzer.go

```diff
@@ -144,10 +145,10 @@ func (a *TcpMetricAnalyzer) getTupleLabels(event *model.KindlingEvent) (*model.A
 	if sIp == nil || sPort == nil || dIp == nil || dPort == nil {
 		return nil, fmt.Errorf("one of sip or dip or dport is nil for event %s", event.Name)
 	}
-	sIpString := model.IPLong2String(uint32(sIp.Value.GetUintValue()))
-	sPortUint := sPort.Value.GetUintValue()
-	dIpString := model.IPLong2String(uint32(dIp.Value.GetUintValue()))
-	dPortUint := dPort.Value.GetUintValue()
+	sIpString := model.IPLong2String(uint32(event.GetUintUserAttribute("sip")))
+	sPortUint := event.GetUintUserAttribute("sport")
+	dIpString := model.IPLong2String(uint32(event.GetUintUserAttribute("dip")))
+	dPortUint := event.GetUintUserAttribute("dport")
```

Comment:
There are several variables above called `sIp`, `sPort`, `dIp`, and `dPort`  which could be reused. I think we better check the value before using it as there are a lot of empty values for these events.